packages/sdk/src/evm/core/classes/contract-publisher.ts

```diff
@@ -370,10 +372,14 @@ export class ContractPublisher extends RPCConnectionHandler {
 
         for (const [network, implementation] of implementationsAddresses) {
           if (implementation !== "") {
+            const compilerMetadata = await fetchContractMetadata(
+              predeployMetadata.metadataUri,
+              this.storage,
```

Comment:
dont do this inside the for loop, its gonna redownload the same thing over and over. just do it under the raw fetch

Reply:
the loop will break as soon as it finds an implementation address.. so this fetch call will only run once

Reply:
talked offline, the loop breaks at first hit so its ok, but should prob still be inside the try/catch